### PR TITLE
Switch to entrypoint-based themes handling

### PR DESCRIPTION
**Pull request by noirbizarre**
This PR switch to entrypoint-based themes handling as it offers many advantages:
- all themes are loaded with the same loader
- themes can now be packaged standalone, no need to be a uData plugin (easier to make a new theme, or a theme template)
- theme is tied to it's distribution, no need anymore to trick to find the real version

There theming process has also been documented.

### CODE REVIEW

File: docs/creating-theme.md [hunk 1]
```diff
@@ -1,11 +1,222 @@
...
+Some images are required and used everywhere in the application:
+
+- `img/flags` contains the language switcher flags in the form `{language code}.png` (ex. `en.png`).
+- `img/placeholders` contains default images for:
+    - users without avatar (`user.png`)
+    - organizations without logo (`organization.png`)
+    - reuses without image (`reuse.png`)
+    - territory without logo or flag (`territory.png`)
+
+Then your free to add any static assets reuired by your theme
```
Comment (davidbgk) [inline docs/creating-theme.md]: reQuired + point at the end

File: docs/creating-theme.md [hunk 2]
```diff
@@ -1,11 +1,222 @@
...
+
+```html+jinja
+{% extends "raw.html" %}
+
+{% block raw_head %}
+{{ super() }}
+<link rel="shortcut icon" href="{{ theme_static('img/favicon.png') }}">
+{% endblock %}
+```
+You can reference static assets from you theme with the `theme_static` global function.
```
Comment (davidbgk) [inline docs/creating-theme.md]: `of your theme`

File: docs/creating-theme.md [hunk 3]
```diff
@@ -1,11 +1,222 @@
...
+
+!!! note
+    Don't forget to call `{{ super() }}` to includes the original block.
```
Comment (davidbgk) [inline docs/creating-theme.md]: include without an S

File: docs/creating-theme.md [hunk 4]
```diff
@@ -1,11 +1,222 @@
...
+    See the [Jinja documentation][] for advanced usage.
+
+You can also rewrite entirely the template, but don't forget you need to have the proper blocks
+for template inheritance and use the proper context variables.
+
+### Hooks
+
+You're theme can also customize some behavior be using hooks in you ``__init__.py``.
```
Comment (davidbgk) [inline docs/creating-theme.md]: YouR + behaviors + bY + youR => `Your theme can also customize some behaviors by using hooks in your ``__init__.py``.`

File: docs/creating-theme.md [hunk 5]
```diff
@@ -1,11 +1,222 @@
...
+Currently there is 2 available hooks:
```
Comment (davidbgk) [inline docs/creating-theme.md]: there ARE

File: docs/creating-theme.md [hunk 6]
```diff
@@ -1,11 +1,222 @@
...
+
+- `theme.menu()` to register a custom main menu
+- `theme.context()` to add extra context variable to some views
+
+You can also expose extras menus using the `udata.app.nav` extension.
+They will be availables in the template context under the `nav` object.
```
Comment (davidbgk) [inline docs/creating-theme.md]: available without an S

File: docs/creating-theme.md [hunk 7]
```diff
@@ -1,11 +1,222 @@
...
+    return context
+```
+
+!!! note
+    You can see an example of advanced hooks usage in the [`gouvfr` theme][gouvfr-hooks].
+
+## Publish and use
+
+Once your theme is ready, you can publish it on [PyPI][] to share it to the world
+(and notify us so we can be glad of your work).
```
Comment (davidbgk) [inline docs/creating-theme.md]: and link to your theme from the documentation?

File: CHANGELOG.md [hunk 8]
```diff
@@ -8,6 +8,9 @@
   [Flask-Security-Fork](https://pypi.python.org/pypi/Flask-Security-Fork)
   [#813](https://github.com/opendatateam/udata/pull/813)
 - Ensure URLs are stripped [#823](https://github.com/opendatateam/udata/pull/823)
+- Themes are now entrypoint-based [#829](https://github.com/opendatateam/udata/pull/829).
+  This is now documented, see [the related documention](https://udata.readthedocs.io/en/stable/creating-theme/)
```
Comment (davidbgk) [inline CHANGELOG.md]: `now` is used twice, I'd remove the second one.

File: docs/creating-theme.md [hunk 9]
```diff
@@ -1,11 +1,222 @@
...
+
+!!! note
+    This is a proposal layout for a standalone theme.
+    As long as the theme package has the proper layout (`info.json`, `__init__.py`...),
+    it can be wherever you want if you properly expose it in your `setup.py` file.
+
+### `setup.py`
+
+The `setup.py` is a classic python `setup.py` file.
+The only requirements is that you properly expose the udata theme packaging
```
Comment (davidbgk) [inline docs/creating-theme.md]: requirement without an S

File: docs/creating-theme.md [hunk 10]
```diff
@@ -1,11 +1,222 @@
...
+}
+```
+The `application` and the `doctype` attributes needs to have specific values, respectively `udata` and `html5`.
+The `identifier` attribute is important: this is the value you will be using in `udata.cfg` to use your theme
+(the `THEME` parameter).
+Any other attribute can have any value, this is only metadata.
+
+### Static assets
+
+The `static` should contains at least 2 files:
```
Comment (davidbgk) [inline docs/creating-theme.md]: contain without an S

File: docs/creating-theme.md [hunk 11]
```diff
@@ -1,11 +1,222 @@
...
+
+- `theme.css` which is loaded automatically on every page
+- `admin.css` which is loaded automatically on on the amdin page
```
Comment (davidbgk) [inline docs/creating-theme.md]: admin (typo on second occurrence)

File: docs/creating-theme.md (hunk 7 above)
Comment (noirbizarre) [inline docs/creating-theme.md]: I added a "Known themes" section with a note on contributions

**Comment by noirbizarre** [comment]
@davidbgk Good to merge ?

File: docs/creating-theme.md [hunk 12]
```diff
@@ -1,11 +1,288 @@
...
+Some images are required and used everywhere in the application:
+
+- `img/flags` contains the language switcher flags in the form `{language code}.png` (ex. `en.png`).
+- `img/placeholders` contains default images for:
+    - users without avatar (`user.png`)
+    - organizations without logo (`organization.png`)
+    - reuses without image (`reuse.png`)
+    - territory without logo or flag (`territory.png`)
+
+Then your free to add any static assets required by your theme.
```
Comment (davidbgk) [inline docs/creating-theme.md]: you are

**Pull request merged**